### PR TITLE
Past medication

### PR DESCRIPTION
**Pull request by kagai**

### CODE REVIEW

File: src/widgets/medications/medication-level-two.component.tsx [hunk 1]
```diff
@@ -50,43 +51,142 @@ export default function MedicationLevelTwo(props: MedicationsOverviewProps) {
...
-          style={{ color: "var(--omrs-color-ink-medium-contrast)" }}
-          className={"omrs-type-body-large"}
+  function displayPastMedications() {
+    return (
+      <React.Fragment>
+        <SummaryCard
+          name={t("Medications - past", "Medications - past")}
+          match={props.match}
+          styles={{ width: "90%" }}
+          addBtnUrl={`/patient/${patientUuid}/chart/add`}
```
Comment (fatmali) [inline src/widgets/medications/medication-level-two.component.tsx]: does this route exist?

File: src/widgets/medications/medication-level-two.component.tsx [hunk 2]
```diff
@@ -50,43 +51,142 @@ export default function MedicationLevelTwo(props: MedicationsOverviewProps) {
...
+                          <td>
+                            <span
+                              style={{
+                                fontWeight: 500,
+                                color: "var(--omrs-color-ink-high-contrast)"
+                              }}
+                            >
+                              {medication.drug.display}
+                            </span>
+                            {" \u2014 "} {medication.doseUnits.display}{" "}
```
Comment (fatmali) [inline src/widgets/medications/medication-level-two.component.tsx]: why so many whitespaces? is there a different way we can achieve this without having the whitespace inside html?

File: src/widgets/medications/medication-level-two.component.tsx [hunk 3]
```diff
@@ -50,43 +51,142 @@ export default function MedicationLevelTwo(props: MedicationsOverviewProps) {
...
+                            <span
+                              style={{
+                                fontWeight: 500,
+                                color: "var(--omrs-color-ink-high-contrast)"
+                              }}
+                            >
+                              {getDosage(
+                                medication.drug.strength,
+                                medication.dose
+                              )}{" "}
```
Comment (fatmali) [inline src/widgets/medications/medication-level-two.component.tsx]: same here about whitespace

File: src/widgets/medications/medication-level-two.component.tsx (hunk 1 above)
Comment (kagai) [reply src/widgets/medications/medication-level-two.component.tsx]: Removed.. it was a place holder thou'

File: src/widgets/medications/medication-level-two.component.tsx (hunk 2 above)
Comment (kagai) [reply src/widgets/medications/medication-level-two.component.tsx]: Removed most of these

File: src/widgets/medications/medication-level-two.component.tsx [hunk 4]
```diff
@@ -31,62 +32,165 @@ export default function MedicationLevelTwo(props: MedicationsOverviewProps) {
...
-            <td>Start Date</td>
-          </tr>
-        </thead>
-        <tbody>{patientMedications && parseMedication()}</tbody>
-      </table>
-      <div className={styles.medicationFooter}>
-        <p
-          style={{ color: "var(--omrs-color-ink-medium-contrast)" }}
-          className={"omrs-type-body-large"}
+  function displayPastMedications() {
```
Comment (drizzentic) [inline src/widgets/medications/medication-level-two.component.tsx]: L35 method has a lot of similarities `displayPastMedication`. Is there a reason we should explicitly have two methods for past and current medication?

Comment (kagai) [reply src/widgets/medications/medication-level-two.component.tsx]: I understand your concern, I tried that at first but it makes the logic complicated, the functions are similar but they have major differences that would make it complicated for no good reason.Let me know your thoughts on this.

**Comment by donaldkibet** [comment]
> _No description provided._

lgtm

**Pull request merged**